app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,18 @@
+class CommentsController < ApplicationController
+  def new
+    @user = current_user
+    @post = Post.all[params[:post_id].to_i - 1]
+    @comment = @user.posts[params[:post_id].to_i - 1].comments.new
+  end
+
+  def create
+    @comment = Comment.create(text: params[:comment][:text], user: current_user, post_id: params[:post_id])
+    if @comment.new_record?
+      redirect_to "/users/#{params[:user_id]}/posts/#{params[:comment][:url_id]}",
+                  flash: { wrong: 'Upps! Comment was not created.' }
+    else
+      redirect_to "/users/#{params[:user_id]}/posts/#{params[:comment][:url_id]}",
+                  flash: { success: 'Comment was successfully created.' }
+    end
+  end
+end
```

Comment:
- [ Optional ] Kindly I recommend you to use `comment_params ` as **private method** .This method is used to handle strong parameters in Rails, which is a security feature that prevents assigning request parameters to objects unless they have been explicitly permitted. For more information you can refer this [Link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.) Just like the below sample 

```
private 
   def comment_params
    params.require(:comment).permit(:text)
  end

```